packages/web-pkg/src/components/AppTemplates/AppWrapper.vue

```diff
@@ -318,19 +331,43 @@ export default defineComponent({
             signal
           })
         }
-        loading.value = false
       } catch (e) {
         console.error(e)
         loadingError.value = e
+      } finally {
         loading.value = false
       }
     }).restartable()
 
     watch(
       currentFileContext,
-      () => {
+      async () => {
         if (!unref(noResourceLoading)) {
-          loadFileTask.perform()
+          await loadResourceTask.perform()
+
+          if (unref(fileSizeLimit) && toNumber(unref(resource).size) > unref(fileSizeLimit)) {
+            dispatchModal({
+              title: $gettext('File %{resource} exceeds %{threshold}', {
```

Comment:
We have file/folder names in the modal title elsewhere as well, but I think mostly with the name at the end of the string so that no information gets lost for long filenames. In this case:

<img width="520" alt="Screenshot 2024-10-08 at 06 26 40" src="https://github.com/user-attachments/assets/0e8a4cb8-7e50-43df-8bff-fb62e961eb5f">

Should we leave out the filename from the title and insert it in the message instead?